README.md

```diff
@@ -1 +1,217 @@
+        }
+      ]
+    }
+  ],
+  actions: [
+    {
+      name: 'add-item',
+      title: 'Add Item',
+      method: 'POST',
+      href: `http://api.x.io/orders/${order.orderNumber}/items`,
+      type: 'application/x-www-form-urlencoded',
+      fields: [
+        { name: 'orderNumber', type: 'hidden', value: `${order.orderNumber}` },
+        { name: 'productCode', type: 'text' },
+        { name: 'quantity', type: 'number' }
+      ]
+    }
+  ],
+  links: [
+    {
+      rel: ['self'],
+      href: `http://api.x.io/orders/${order.orderNumber}`
+    },
+    {
+      rel: ['previous'],
+      href: `http://api.x.io/orders/${order.orderNumber - 1}`
+    },
+    {
+      rel: ['next'],
+      href: `http://api.x.io/orders/${order.orderNumber + 1}`
+    }
+  ]
+});
+```
+
+If you don't want to build your entity in one fell swoop (I wouldn't blame you),
+you can use the other component classes: `Action`, `Field`, `Link`,
+`EmbeddedEntity`, and `EmbeddedLink`. Each of these accept required
+members as positional constructor arguments and optional members in a final
+options object.
+
+Here's how you might break up the above code:
+
+```js
+const orderUrl = `http://api.x.io/orders/${order.orderNumber}`;
+const selfLink = new Siren.Link(['self'], orderUrl);
+
+const itemsRel = 'http://x.io/rels/order-items';
+const itemsUrl = `http://api.x.io/orders/${order.orderNumber}/items`;
+const itemsLink = new Siren.EmbeddedLink([itemsRel], itemsUrl, {
+  class: ['items', 'collection']
+});
+
+const customerRel = 'http://x.io/rels/customer';
+const customerEntity = new Siren.EmbeddedEntity([customerRel], {
+  class: ['info', 'customer'],
+  properties: {
+    customerId: order.customer.userId,
+    name: order.customer.fullName
+  },
+  links: [
+    {
+      rel: ['self'],
+      href: `http://api.x.io/customers/${order.customer.userId}`
+    }
+  ]
+});
+
+const quantityField = new Siren.Field('quantity', {
+  type: 'number'
+});
+
+const addItemAction = new Siren.Action('add-item', itemsUrl, {
+  title: 'Add Item',
+  method: 'POST',
+  type: 'application/x-www-form-urlencoded',
+  fields: [
+    { name: 'orderNumber', type: 'hidden', value: `${order.orderNumber}` },
+    { name: 'productCode', type: 'text' },
+    { name: 'quantity', type: 'number' }
+  ]
+});
+```
+
+Now constructing the full entity is a little easier.
+
+```js
+new Siren.Entity({
+  class: ['order'],
+  properties: {
+    orderNumber: order.orderNumber,
+    itemCount: order.items.length,
+    status: order.orderStatus
+  },
+  entities: [itemsLink, customerEntity],
+  actions: [addItemAction],
+  links: [
+    selfLink,
+    {
+      rel: ['previous'],
+      href: `http://api.x.io/orders/${order.orderNumber - 1}`
+    },
+    {
+      rel: ['next'],
+      href: `http://api.x.io/orders/${order.orderNumber + 1}`
+    }
+  ]
+});
+//=> same as entity
+```
+
+### Generating Siren
+
+To generate a Siren representation, use `JSON.stringify()`.
```

Comment:
"To generate a string representation of a Siren entity [...]"?

Reply:
Not sure I see the difference, semantically. Could you elaborate your thought with that change?